packages/datatrak-web/src/AppProviders.tsx

```diff
@@ -12,23 +12,29 @@ import { theme } from './theme';
 import { Toast } from './components';
 import { errorToast } from './utils';
 import { CurrentUserContextProvider } from './api';
+import { REDIRECT_ERROR_PARAM } from './constants';
+
+const handleError = (error: any, query: any) => {
+  if (error.responseData.redirectClient) {
+    // Redirect the browser to the specified URL and display the error
+    window.location.href = `${error.responseData.redirectClient}?${REDIRECT_ERROR_PARAM}=${error.message}`;
```

Comment:
This refreshes the tab which is why we need to pass through the error as a url param. I spent a good while trying to get it to work by navigating to `/login` with `react-router-dom` but it was proving quite tricky. Ultimately I think I prefer this solution anyway since it really gives the sense of the user being booted out and forced to login again.

Reply:
Out of interest, what went wrong here with using `react-router-dom`?

Reply:
I think in pre-v6 you could directly manipulate the browser history using `window.history.pushState({}, '', url)`. But with v6 and the use of `BrowserRouter` it uses its own history, so you can't directly change it outside of a component.

Since we want the error handling to come from the `QueryClient` which is defined outside of a component, it was a bit awkward trying to inject in a `useNavigate()` reference.

I was able to do that in the end, but even then it didn't seem to be navigating to `/login`. There's probably a bit more work that could be done to work out exactly this is the case, but I think I was finding other issues due to state not being cleared. Forcing a tab refresh felt a lot more reliable as a way of kicking the user out.

Reply:
I think maybe because the `CurrentUserContext` still had my user? So we kept getting sent back to the home screen by the `AuthViewLoggedInRedirect`?